hardware/bsg_axi_bus_pkg.vh

```diff
@@ -57,85 +61,155 @@ typedef struct packed { \
   \
 } miso_struct_name
 
-`define bsg_axil_mosi_bus_width(slot_num_lp) \
-( slot_num_lp * (3*32 + 5 + 4) \
+`define bsg_axil_mosi_bus_width(slot_num_mp) \
+( slot_num_mp * (3*32 + 5 + 4) \
 )
 
-`define bsg_axil_miso_bus_width(slot_num_lp) \
-( slot_num_lp * (5 + 2*2 + 32) \
+`define bsg_axil_miso_bus_width(slot_num_mp) \
+( slot_num_mp * (5 + 2*2 + 32) \
 )
 
 
-`define declare_bsg_axi_bus_s(slot_num_lp, id_width_p, addr_width_p, data_width_p, mosi_struct_name, miso_struct_name) \
+// -----------------------------------------------------------
+// AXI4 (simple version)
+// -----------------------------------------------------------
+`define declare_bsg_axi_bus_s(slot_num_mp, id_width_mp, addr_width_mp, data_width_mp, mosi_struct_name, miso_struct_name) \
```

Comment:
regarding this. I think it might be better to have slot_num_mp outside. If you want to make an array of axi ports, I think it's just better to do something like.
```
mosi_struct_name [slot_num_p-1:0] mosi_ports;

```

Also why do we need struct_name as macro parameter?

Reply:
That is a good point, and much simpler!

However, we do need some concatenated interface defined above, for the Xilinx IPs, such as crossbar...

For the second question, we need the macro name to differentiate AXI interfaces with different parameters, like the id_width_mp. It is possible that we use these AXI4 interfaces in one module.

Reply:
Are you saying that the ports on the xilinx IPs are defined something like,
so that you would need to transpose the array if defined the other way?

```
module xilinx_ip #(parameter slot_num_p="inv", parameter id_width_p="inv")
(
  input [slot_num_mp-1:0][id_width_mp-1:0] awid
);

endmodule
```
well in that case, this might be better.

Reply:
Yes, exactly.